recipes/nameof/all/conanfile.py

```diff
@@ -0,0 +1,58 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class NameofConan(ConanFile):
+    name = "nameof"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Neargye/nameof"
+    license = "MIT"
+    description = "Header-only C++17 library provides nameof macros and functions to simply obtain the name of a variable, type, function, macro, and enum."
+    topics = (
+        "conan",
+        "nameof",
+        "cplusplus",
+        "enum-to-string",
+        "serialization",
+        "reflection",
+        "header-only",
+        "compile-time"
+    )
+    no_copy_source = True
+    settings = "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _supported_compiler(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if compiler == "Visual Studio" and version >= "15":
+            return True
+        if compiler == "gcc" and version >= "7":
+            return True
+        if compiler == "clang" and version >= "5":
+            return True
+        if compiler == "apple-clang" and version >= "9":
+            return True
+        return False
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if not self._supported_compiler:
```

Comment:
In case someone uses the `compiler.cppstd` subsetting is a good practice to let him know that 
we require a minimum.

```suggestion
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, "17")
        if not self._supported_compiler:
```
Merged example here:

https://github.com/conan-io/conan-center-index/blob/d3c8a2180c4b5e95f63c08c518e1b1a9a7017a23/recipes/magic_enum/all/conanfile.py#L46